Sources/DolbyIORTSUIKit/Private/ViewModels/ErrorViewModel.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ErrorViewModel.swift
+//
+    
+import DolbyIORTSCore
+import Foundation
+
+final class ErrorViewModel {
+
+    let titleText: String
+    let subtitleText: String?
+
+    init(titleText: String, subtitleText: String?) {
+        self.titleText = titleText
+        self.subtitleText = subtitleText
+    }
+    
+    init(error: Error) {
+        let titleAndSubtitle = Self.titleAndSubtitleForError(error)
+        self.titleText = titleAndSubtitle.title
+        self.subtitleText = titleAndSubtitle.subtitle
+    }
+    
+    private typealias ErrorTitleAndSubtitle = (title: String, subtitle: String?)
+    private static func titleAndSubtitleForError(_ error: Error) -> ErrorTitleAndSubtitle {
+        switch error {
+        case let streamError as StreamError:
+            switch streamError {
+            case .connectFailed:
+                return (.offlineErrorTitle, .offlineErrorSubtitle)
+            default:
+                return (.genericErrorTitle, nil)
+            }
+        default:
+            return (.genericErrorTitle, nil)
+        }
+    }
+}
+
+extension ErrorViewModel {
```

Comment:
Suggestion, use error model

Reply:
`StreamError` already handled here is the only error object we have for now, however the init method takes in any error type that conforms to `Error` protocol.